ee/desktop/menu/menu_template.go

```diff
@@ -0,0 +1,89 @@
+package menu
+
+import (
+	"fmt"
+	"strings"
+	"text/template"
+
+	"github.com/kolide/kit/version"
+)
+
+type TemplateDataOption func(*templateData)
+
+func WithVersion(info version.Info) TemplateDataOption {
+	return func(td *templateData) {
+		td.LauncherVersion = info.Version
+		td.LauncherRevision = info.Revision
+		td.GoVersion = info.GoVersion
+	}
+}
+
+func WithOSQueryVersion(osqueryVersion string) TemplateDataOption {
+	return func(td *templateData) {
+		td.OSQueryVersion = osqueryVersion
+	}
+}
+
+func WithHostname(hostname string) TemplateDataOption {
+	return func(td *templateData) {
+		td.Hostname = hostname
+	}
+}
+
+func WithLogFilePath(logFilePath string) TemplateDataOption {
+	return func(td *templateData) {
+		td.LogFilePath = logFilePath
+	}
+}
+
+func WithLauncherFlagsFilePath(launcherFlagsFilePath string) TemplateDataOption {
+	return func(td *templateData) {
+		td.LauncherFlagsFilePath = launcherFlagsFilePath
+	}
+}
+
+type templateData struct {
+	LauncherVersion       string
+	LauncherRevision      string
+	GoVersion             string
+	OSQueryVersion        string
+	Hostname              string
+	LogFilePath           string
+	LauncherFlagsFilePath string
+}
+
+func NewTemplateData(opts ...TemplateDataOption) *templateData {
```

Comment:
I'm not totally sure about the `opts` pattern for this. It feels okay for things that we know during the initial `main` setup, and don't change, but I'm not sure that's true for this.

Though, even things we know at compile time, I might just bake into that `templateData` array, instead of the extra pluming to pass `With...` along.

For things that we don't know... We need to generate them after startup. I did something similar in `localserver`. It has a `Querier`, and a routine that polls it to update the internal data structure. Poke around https://github.com/kolide/launcher/blob/main/ee/localserver/request-id.go#L35

Reply:
When I say that, it all sounds really complicated. So maybe we just hardcode the things we know at compile time and expand later?

Reply:
Yeah I was on the fence with this pattern too - I have to make a larger change to how template data is fed to desktop, and I think that will make this go away.

Reply:
This has been simplified and I ripped this stuff out!